### DISCUSSION TITLE
How could private likes work in atproto?

### DISCUSSION THREAD

**Discussion by beholdnec** [bluesky-social/atproto#1983] [Protocol (atproto)]
I have no specific plans or expertise here, but I'm wondering what a "private like" system would look like on Bluesky.

There would have to be a system where a user could transmit a verified, unspoofable fingerprint that verifies a real user liked a post, yet the fingerprint would have to be untraceable to the original user.

I imagine the user would have to belong to a verified group of users, and the fingerprint would signify, "A user from group X liked this post, and you can verify the user came from group X, but you cannot tell which user it was."

The verified group is needed because otherwise there would be no way to tell whether the fingerprint belongs to a real user or a spam bot.

I'm just pondering about the problem. What do you think?

**Comment by snarfed** [comment]
I'd be curious to hear more about the use case here. I'm not familiar with many (any?) other existing systems where you can "private like" something like this. Do you know of any precedents? A number of social networks do something like this when you have a private/protected _account_, but that setting is at the account level, not per like.

Polls (#1310) are maybe a better example. Many (most?) social networks with polls make individual votes/responses private.

**Reply by pfrazee** [reply]
This all gets into the territory of [e2e auditable voting](https://en.wikipedia.org/wiki/End-to-end_auditable_voting_systems). I'd have to refresh my memory on the current state-of-the-art, but my recollection is that the zero-trust schemes involve homomorphic encryption (for tabulation) and/or zk proof schemes, and most of that cryptography is still pretty novel, especially if it's non-interactive. You also have to ask what the merit of auditability is when you're running open-participation votes due to the sybil problem.

For polls, I think the best choice is to lean on reputable services to keep the votes private and the results consistent. There might be some interactive secret ballot models you could use there, but ideally someone who is an expert in these systems would get involved.
